atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClient.java

```diff
@@ -75,22 +93,58 @@ public void batch_mutate(Map<ByteBuffer, Map<String, List<Mutation>>> mutation_m
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
         checkLimitAndCall(() -> {
             super.batch_mutate(mutation_map, consistency_level);
+
+            qosMetrics.updateWriteCount();
+            qosMetrics.updateBytesWritten(getApproximateWriteByteCount(mutation_map));
             return null;
         });
     }
 
+    private int getApproximateWriteByteCount(Map<ByteBuffer, Map<String, List<Mutation>>> mutation_map) {
+        int approxBytesForKeys = mutation_map.keySet().stream().mapToInt(e -> e.array().length).sum();
+        int approxBytesForValues = mutation_map.values().stream()
+                .mapToInt(singleMap -> {
+                    int approximateBytesInStrings = singleMap.keySet().stream().mapToInt(String::length).sum();
+                    int approximateBytesInMutations = singleMap.values().stream()
+                            .mapToInt(listOfMutations -> listOfMutations.stream()
+                                    .mapToInt(mutation -> SerializationUtils.serialize(mutation).length)
```

Comment:
again would prefer not serializing if we can avoid it

Reply:
A mutation again has:

  public ColumnOrSuperColumn column_or_supercolumn; // optional
  public Deletion deletion; // optional

We can resuse the ColumnOrSuperColumn calculation here, will be happy to do if its better.